schema.py

```diff
@@ -266,6 +267,26 @@ def is_valid(self, data):
         else:
             return True
 
+
+    def set_schema_name(self, error_type):
+        """Returns string to add to error message based on what the error type
+        is and whether a custom schema name has been defined or not.
+        """
+        if self._name != '':
+            if error_type == "SchemaError":
+                error_message = "\'%s\' " % (self._name,)
```

Comment:
I think @rmorshea is referring to the `%r`, which you can use to get a representation of the string, so you don't have to add the quotes yourself. Basically, this whole section can be:

```
if error_type in ("SchemaError", "MissingKey", "WrongKey", "UnexpectedType":
    return repr(self._name)
```

Reply:
Right now, since all the schema names are formatted the same way for each error type
`"\'%s\' " % (self._name,)`
it is redundant. But I was thinking the additional flexibility of being able to make the schema names' formatting unique to the error type it's appended to could be beneficial since some of the schema error messages that get raised can vary a bit depending on the the type of schema error it is

Reply:
Sure, but this is easy to change later, so YAGNI dictates that we should go with the simplest implementation for now.

Reply:
Now the way `self._raise_error()` operates, by taking in the "error_type" (now renamed to "exception"), "message", and "error_list", it makes checking the "error_type" redundant making the function itself simpler and shorter 👍.

Also `self._name` is now formatted with `repr` instead of `str`